### PR TITLE
Allow list of hatches to {bar, barh}

### PR DESCRIPTION
**Pull request by dstansby**
## PR Summary
Fixes https://github.com/matplotlib/matplotlib/issues/17758. This seemed pretty easy. I think there's room to add some helper functions to do the broadcasting and iterating for all of the rect properties, but perhaps that can be done elsewhere? First PR in a while, so apologies if I've got something wrong...

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by dstansby** [comment]
Good idea - I updated that example and it's quite a bit better without having to loop through the bar patches individually.

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -1551,6 +1551,20 @@ def test_bar_pandas_indexed(pd):
     ax.bar(df.x, 1., width=df.width)
 
 
+@check_figures_equal()
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py:1554]:
```suggestion
@check_figures_equal()
@pytest.mark.stile('default')
```

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -1551,6 +1551,20 @@ def test_bar_pandas_indexed(pd):
...
+def test_bar_hatches(fig_test, fig_ref):
+    ax_test = fig_test.subplots()
+    ax_ref = fig_ref.subplots()
+
+    x = [1, 2]
+    y = [2, 3]
+    hatches = ['x', 'o']
+    for i in range(2):
+        ax_ref.bar(x[i], y[i], color='blue', hatch=hatches[i])
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py:1563]:
```suggestion
        ax_ref.bar(x[i], y[i], color='C0', hatch=hatches[i])
```

**Comment by tacaswell** [comment]
Also, welcome back @dstansby  :)

**Pull request merged**